### ISSUE TITLE
 The "reloading tests" round arrow wont stop

### ISSUE THREAD

**Issue by matepek** [hbenl/vscode-mocha-test-adapter#45]
I was trying to debug my extension (in an environment where your mocha was installed).
The "reloading tests" round arrow wont stop.
After I deleted your mocha one, it works fine.

Repro(I guess):
- Install mocha test extension
- Try debug my extension / maybe just installing is enough.

```
rejected promise not handled within 1 second: TypeError: Cannot read property 'stack' of undefined
extensionHostProcess.js:700
stack trace: TypeError: Cannot read property 'stack' of undefined
extensionHostProcess.js:700
	at MochaAdapter.<anonymous> (/Users/matepek/.vscode/extensions/hbenl.vscode-mocha-test-adapter-1.7.0/out/core.js:85:78)
	at Generator.throw (<anonymous>)
	at rejected (/Users/matepek/.vscode/extensions/hbenl.vscode-mocha-test-adapter-1.7.0/node_modules/tslib/tslib.js:105:69)
	at process._tickCallback (internal/process/next_tick.js:68:7)
```

**Comment by matepek** [comment]
Node: "Error while loading tests - click to show"

Output window content:
> undefined

But at least now my extension works fine next to it.

**Comment by hbenl** [comment]
> Output window content of the node [...]

Yeah, that's what I expected - the original issue was that I didn't guard against getting `undefined` as the error in a catch clause.
I'll still have to find out who does a `throw undefined` and why.

**Comment by matepek** [comment]
Here:
https://github.com/hbenl/vscode-mocha-test-adapter/blob/b3b096a3ae95d8d8b300871cb7c85ffd02c4e029/src/optsReader.ts#L141-L145

Stack is like:
```
new Error().stack
"Error: 
	at eval (eval at <anonymous> (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/optsReader.js:62:29), <anonymous>:1:1)
	at MochaOptsReader.<anonymous> (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/optsReader.js:62:29)
	at Generator.next (<anonymous>)
	at __awaiter (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/tslib/tslib.js:107:75)
	at new Promise (<anonymous>)
	at Object.__awaiter (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/tslib/tslib.js:103:16)
	at MochaOptsReader.readOptsUsingMocha (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/optsReader.js:61:24)
	at ConfigReader.<anonymous> (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/configReader.js:64:50)
	at Generator.next (<anonymous>)
	at __awaiter (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/tslib/tslib.js:107:75)
	at new Promise (<anonymous>)
	at Object.__awaiter (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/tslib/tslib.js:103:16)
	at ConfigReader.readConfig (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/configReader.js:53:24)
	at new ConfigReader (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/configReader.js:14:36)
	at new MochaAdapter (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/adapter.js:15:29)
	at TestAdapterRegistrar.tslib_1.__awaiter.context.subscriptions.push.vscode_test_adapter_util_1.TestAdapterRegistrar [as adapterFactory] (/Volumes/Secure/repo/vscode-mocha-test-adapter/out/main.js:18:122)
	at TestAdapterRegistrar.add (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/vscode-test-adapter-util/out/registrar.js:41:30)
	at new TestAdapterRegistrar (/Volumes/Secure/repo/vscode-mocha-test-adapter/node_modules/vscode-test-adapter-util/out/registrar.js:19:22)
	at /Volumes/Secure/repo/vscode-mocha-test-adapter/out/main.js:18:40
```

**Comment by hbenl** [comment]
D'oh! That was stupid. It's fixed in 1.7.2.

I'm also going to add some logic soon that disables the adapter for workspaces that show no indication of using mocha and (in that case) ask the user if he wants to enable it if the workspace contains files matching `test/**/*.js`. This has been on my TODO list anyway...

**Comment by hbenl** [comment]
You can now disable the adapter for any workspace folder. If the adapter isn't sure if it should be enabled for a folder (because there are test files but no indication if they should be run using mocha), it will ask if you want to enable it for that folder.
